File: Medium/2025-12-15-2110-Number-of-Smooth-Descent-Periods-of-a-Stock/solution.java
class Solution {
    public long getDescentPeriods(int[] prices) {
        long count = 0;
        int n = prices.length;
        int currentPeriod = 1;

        for (int i = 1; i < n; i++) {
            if (prices[i] == prices[i - 1] - 1) {
                currentPeriod++;
            } else {
                currentPeriod = 1;
            }
            count += currentPeriod;
        }
        
        return count + n; // Add n for the single-day periods
    }
}

File: Medium/2025-12-15-2110-Number-of-Smooth-Descent-Periods-of-a-Stock/solution.py
class Solution:
    def getDescentPeriods(self, prices: List[int]) -> int:
        total_periods = len(prices)
        current_period = 0
        
        for i in range(1, len(prices)):
            if prices[i] == prices[i - 1] - 1:
                current_period += 1
            else:
                current_period = 0
            total_periods += current_period
        
        return total_periods